engine/src/configuration/applier/serviceescalation.cc

```diff
@@ -187,9 +190,10 @@ void applier::serviceescalation::remove_object(
   } else
     service_exists = true;
 
+  size_t key = serviceescalation_key(obj);
   for (serviceescalation_mmap::iterator it{range.first}, end{range.second};
        it != end; ++it) {
-    if (it->second->get_uuid() == obj.uuid()) {
+    if (it->second->internal_key() == key) {
```

Comment:
event if keys are equals, objects may be differents, a first compare of key and then a compare of objects would be better.